sdkproject/Assets/Mapbox/Core/Unity/MeshGeneration/Factories/TerrainFactory.cs

```diff
@@ -43,6 +43,12 @@ public class TerrainFactory : Factory
         private float _heightModifier = 1f;
         [SerializeField]
         private int _sampleCount = 40;
+        [SerializeField]
+        private bool _addMeshCollider = false;
+        [SerializeField]
+        private bool _addToLayer = false;
+        [SerializeField]
+        private int _layerId = 0;
```

Comment:
Should the layer option be in the base class? It makes sense for all factories to have the ability to specify which layer they place objects in. Further down the chain, specific stacks can offer layer overrides?